python_modules/dagster/dagster/_core/definitions/assets_lazy.py

```diff
@@ -27,27 +28,41 @@ class AssetsDefinitionMetadata(
     def __new__(
         cls,
         input_keys: AbstractSet[AssetKey],
-        output_keys: AbstractSet[AssetKey],
-        asset_deps: Optional[Dict[AssetKey, AbstractSet[AssetKey]]] = None,
+        output_keys: Mapping[str, AssetKey],
+        asset_deps: Optional[Dict[str, AbstractSet[AssetKey]]] = None,
         group_names_by_key: Optional[Dict[AssetKey, str]] = None,
+        metadata: Optional[MetadataUserInput] = None,
         metadata_by_key: Optional[Dict[AssetKey, MetadataUserInput]] = None,
     ):
         return super().__new__(
             cls,
             input_keys=check.set_param(input_keys, "input_keys", of_type=AssetKey),
-            output_keys=check.set_param(output_keys, "output_keys", of_type=AssetKey),
+            output_keys=check.dict_param(
+                output_keys, "output_keys", key_type=str, value_type=AssetKey
+            ),
             asset_deps=check.opt_nullable_dict_param(
-                asset_deps, "asset_deps", key_type=AssetKey, value_type=set
+                asset_deps, "asset_deps", key_type=str, value_type=set
             ),
             group_names_by_key=check.opt_nullable_dict_param(
                 group_names_by_key, "group_names_by_key", key_type=AssetKey, value_type=str
             ),
+            metadata=metadata,
             metadata_by_key=check.opt_nullable_dict_param(
                 metadata_by_key, "metadata_by_key", key_type=AssetKey, value_type=MetadataUserInput
             ),
         )
 
 
+@whitelist_for_serdes
```

Comment:
just needed so we can save it as a snapshot, since we can't pass a list directly